src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp

```diff
@@ -0,0 +1,138 @@
+  }
+  if (abs(angular_velocity) >= 1.0) {
+    PARSE_ERROR(context, "The rotation angular velocity ("
+                             << angular_velocity_
+                             << ") must be between -1.0 and 1.0");
+  }
+  if ((tilt_angle < 0.0) or (tilt_angle > M_PI)) {
+    PARSE_ERROR(context, "The rotator tilt angle ("
+                             << tilt_angle_ << ") must be between 0 and Pi");
+  }
+}
+
+RotatingDipole::RotatingDipole(CkMigrateMessage* msg) : InitialData(msg) {}
+
+std::unique_ptr<evolution::initial_data::InitialData>
+RotatingDipole::get_clone() const {
+  return std::make_unique<RotatingDipole>(*this);
+}
+
+void RotatingDipole::pup(PUP::er& p) {
+  InitialData::pup(p);
+  p | vector_potential_amplitude_;
+  p | varpi0_;
+  p | delta_;
+  p | angular_velocity_;
+  p | tilt_angle_;
+  p | background_spacetime_;
+}
+
+PUP::able::PUP_ID RotatingDipole::my_PUP_ID = 0;
+
+tuples::TaggedTuple<Tags::TildeE> RotatingDipole::variables(
+    const tnsr::I<DataVector, 3>& coords, tmpl::list<Tags::TildeE> /*meta*/) {
+  return {make_with_value<tnsr::I<DataVector, 3>>(coords, 0.0)};
+}
+
+tuples::TaggedTuple<Tags::TildeB> RotatingDipole::variables(
+    const tnsr::I<DataVector, 3>& coords,
+    tmpl::list<Tags::TildeB> /*meta*/) const {
+  auto result = make_with_value<tnsr::I<DataVector, 3>>(coords, 0.0);
+
+  const double sin_alpha = sin(tilt_angle_);
+  const double cos_alpha = cos(tilt_angle_);
+
+  // Coordinates and magnetic fields in the tilted axis
+  const auto& x = get<0>(coords);
+  const auto& y = get<1>(coords);
+  const auto& z = get<2>(coords);
+  const DataVector x_prime = cos_alpha * x - sin_alpha * z;
+  const DataVector z_prime = sin_alpha * x + cos_alpha * z;
+
+  auto tilde_b_prime = make_with_value<tnsr::I<DataVector, 3>>(coords, 0.0);
+
+  // Regularized dipole field
+  const DataVector r_squared = get(dot_product(coords, coords));
+  const DataVector one_over_radius_factor =
+      1.0 / pow<5>(sqrt(r_squared + square(delta_)));
+  get<0>(tilde_b_prime) = 3.0 * x_prime * z_prime * one_over_radius_factor;
+  get<1>(tilde_b_prime) = 3.0 * y * z_prime * one_over_radius_factor;
+  get<2>(tilde_b_prime) =
+      (3.0 * square(z_prime) - r_squared + 2.0 * square(delta_)) *
+      one_over_radius_factor;
+
+  // Rotation
+  get<0>(result) =
+      cos_alpha * get<0>(tilde_b_prime) + sin_alpha * get<2>(tilde_b_prime);
+  get<1>(result) = get<1>(tilde_b_prime);
+  get<2>(result) =
+      -sin_alpha * get<0>(tilde_b_prime) + cos_alpha * get<2>(tilde_b_prime);
+
+  return result;
+}
+
+tuples::TaggedTuple<Tags::TildePsi> RotatingDipole::variables(
+    const tnsr::I<DataVector, 3>& coords, tmpl::list<Tags::TildePsi> /*meta*/) {
+  return {make_with_value<Scalar<DataVector>>(coords, 0.0)};
+}
+
+tuples::TaggedTuple<Tags::TildePhi> RotatingDipole::variables(
+    const tnsr::I<DataVector, 3>& coords, tmpl::list<Tags::TildePhi> /*meta*/) {
+  return {make_with_value<Scalar<DataVector>>(coords, 0.0)};
+}
+
```

Comment:
If you were asking about the NS interior, we're overwriting `Q=0` inside the NS so this would be fine. If overwriting Q=0 inside the NS is something we shouldn't do, this should be fixed as you pointed out.

Reply:
For the evolution that's totally fine, because Q decouples from the E and B evolution with the fixing.

Maybe add a comment on this, too?